src/lib/custom-resources/cdk-s3-update-logarchive-bucket-policy/runtime/src/index.ts

```diff
@@ -0,0 +1,201 @@
+import * as AWS from 'aws-sdk';
+AWS.config.logger = console;
+import {
+  CloudFormationCustomResourceEvent,
+  CloudFormationCustomResourceCreateEvent,
+  CloudFormationCustomResourceUpdateEvent,
+  CloudFormationCustomResourceDeleteEvent,
+} from 'aws-lambda';
+import { errorHandler } from '@aws-accelerator/custom-resource-runtime-cfn-response';
+
+export interface HandlerProperties {
+  roles: string[];
+  logBucketArn: string;
+  logBucketName: string;
+  logBucketKmsKeyArn: string | undefined;
+}
+
+export const handler = errorHandler(onEvent);
+
+const kms = new AWS.KMS();
+const s3 = new AWS.S3();
+const logArchiveReadOnlySid = 'SSM Log Archive Read Only Roles';
+
+interface PolicyStatement {
+  Sid: string;
+  Effect: string;
+  Action: string[];
+  Principal: {
+    AWS: string[];
+  };
+  Resource: string[] | string;
+}
+
+async function onEvent(event: CloudFormationCustomResourceEvent) {
+  console.log(`Adding roles with /'ssm-log-archive-read-only-access: true/'
+   to the Log Archive Bucket Policy...`);
+
```

Comment:
`console.log(JSON.stringify(event, null, 2));`

We are currently writing input into clowdwatch to quickly debug if we get any issue.

Reply:
Added this change in latest commit